src/recipe/recipe.controller.ts

```diff
@@ -65,9 +75,13 @@ export class RecipeController {
   async fetchRecipe(
     @Param('id', ParseIntPipe) id: number,
   ): Promise<FetchRecipeResponse> {
-    const fetchedRecipe = await this.recipeService.fetchRecipe(id);
+    let recipe: Recipe = await this.redisCacheService.get(`${id}`);
```

Comment:
Well, just like mentioned above, you're storing the recipe in the cache under `cache:recipe:${createdRecipe.id}` key, but try to read it from `${id}` key.

Use the `recipeCacheKey` mentioned above in all places.

Reply:
You could also create a special type using [Template Literal Types](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html) to use the type system to guard you against such issues.

[Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAShDGBLSBhAhvAFhA0hEAPHEpAJIAmUEAHsBAHbkDOUTwATovQOZQC8rDl24A+flAAG8DNgBc7BMgiyAJAG9iSigF8JAbgBQBgGYBXevGCIA9vSjcIwTZABi7awFt0WCAAoA1viysIqoMrj4AJTB5v701gDudmoGUFAKwKbsydqG2kYOTqEQbp7e2L4A5NI+8sWyaEzklZGGhc4l7l7hVY3NrUA)

This way, you also won't be able to e.g. overwrite recipe data with user data, because the key won't have proper structure.